include/boost/geometry/algorithms/detail/disjoint/areal_areal.hpp

```diff
@@ -129,6 +179,67 @@ struct areal_areal
     }
 };
 
+template <typename Areal, typename Box>
+struct areal_box_with_info
+{
+    typedef typename point_type<Areal>::type point_type;
+    typedef segment_intersection_points<point_type> intersection_return_type;
+
+    template <typename Strategy>
+    static inline intersection_return_type
+    apply(Areal const& areal,
+          Box const& box,
+          Strategy const& strategy)
+    {
+        intersection_return_type res =
+                for_each_segment(geometry::segments_begin(areal),
+                                 geometry::segments_end(areal),
+                                 box, strategy);
+                                 //strategy.get_disjoint_segment_box_strategy());
+        if ( res.count != 0 )
+        {
+            return res;
+        }
+
+        // If there is no intersection of any segment and box,
+        // the box might be located inside areal geometry
+
+        if ( point_on_border_covered_by(box, areal,
+                strategy.template get_point_in_geometry_strategy<Box, Areal>()) )
+        {
+            res.count = 1;
+            res.intersections[0] = box.max_corner();
+            return res;
+        }
+
+        return intersection_return_type();
+    }
+
+private:
+    template <typename SegIter, typename Strategy>
+    static inline intersection_return_type
+    for_each_segment(SegIter first,
+                     SegIter last,
```

Comment:
More proper name would be something like `check_segments_range`. An algorithm `for_each_segment` would call a predicate for each segment of a geometry, like that: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/for_each.hpp#L366

Reply:
Again this is derived from original implementation of distance. But I agree, so I changed the name of both.